src/commonMain/kotlin/fr/acinq/lightning/wire/LightningCodecs.kt

```diff
@@ -225,4 +224,26 @@ object LightningCodecs {
         return bytes(input, length)
     }
 
+    fun encodedNodeId(input: Input): EncodedNodeId {
+        val firstByte = byte(input)
+        if (firstByte == 0 || firstByte == 1) {
+            val isNode1 = firstByte == 0
+            val scid = ShortChannelId(int64(input))
+            return EncodedNodeId.ShortChannelIdDir(isNode1, scid)
+        } else if (firstByte == 2 || firstByte == 3) {
+            val publicKey = PublicKey(ByteArray(1) { firstByte.toByte() } + bytes(input, 32))
+            return EncodedNodeId.Plain(publicKey)
+        } else {
+            throw IllegalArgumentException("unexpected first byte: $firstByte")
```

Comment:
Note that you'll need to be careful when decrypting an onion message and decoding its content to catch that exception.